pkg/cri/resource-manager/cache/pod.go

```diff
@@ -109,6 +111,32 @@ func (p *pod) GetContainers() []Container {
 	return containers
 }
 
+// Get container pointer by its name.
+func (p *pod) getContainer(name string) *container {
+	var found *container
+
+	if p.containers == nil {
+		p.containers = make(map[string]string)
+	}
+
+	for _, c := range p.GetContainers() {
+		cptr := c.(*container)
+		p.containers[cptr.Name] = cptr.ID
+		if cptr.Name == name {
+			found = cptr
+		}
+	}
```

Comment:
Hmm... this looks wrong. Seems like the idea was to cache successful lookups on the slow path, but only the slow-path has been now implemented. The fast-path (else branch) is missing altogether... I'll fix this as well.

Reply:
Fixed.